components/integrations_marketplace/marketplace_item/marketplace_item.jsx

```diff
@@ -0,0 +1,186 @@
+import ConfirmModal from 'components/confirm_modal.jsx';
+import PluginIcon from 'components/widgets/icons/plugin_icon.jsx';
+
+import {localizeMessage} from 'utils/utils';
+
+export default class MarketplaceItem extends React.Component {
+        static propTypes = {
+            id: PropTypes.string.isRequired,
+            name: PropTypes.string.isRequired,
+            description: PropTypes.string.isRequired,
+            version: PropTypes.string.isRequired,
+            downloadUrl: PropTypes.string,
+            homepageUrl: PropTypes.string,
+            installed: PropTypes.bool.isRequired,
+            onConfigure: PropTypes.func.isRequired,
+            onInstalled: PropTypes.func.isRequired,
+            actions: PropTypes.shape({
+                installPluginFromUrl: PropTypes.func.isRequired,
+            }).isRequired,
+        };
+
+        constructor(props) {
+            super(props);
+
+            this.state = {
+                installed: this.props.installed,
+                installing: false,
+                confirmOverwriteInstallModal: false,
+                serverError: null,
+            };
+        }
+
+        onClickOvewriteModal = () => {
+            this.setState({confirmOverwriteInstallModal: false});
+            return this.installPlugin(true);
+        }
+
+        onCancelOvewriteModal = () => {
+            this.setState({
+                confirmOverwriteInstallModal: false,
+                installing: false,
+                serverError: null,
+            });
+        }
+
+        installPlugin = async (force) => {
+            const {error} = await this.props.actions.installPluginFromUrl(this.props.downloadUrl, force);
+
+            if (error) {
+                if (error.server_error_id === 'app.plugin.install_id.app_error' && !force) {
+                    this.setState({confirmOverwriteInstallModal: true});
+                    return;
+                }
+
+                this.setState({
+                    installing: false,
+                });
+
+                //WIP: Error handling
+
+                if (error.server_error_id === 'app.plugin.extract.app_error') {
+                    this.setState({serverError: localizeMessage('admin.plugin.error.extract', 'Encountered an error when extracting the plugin.')});
+                } else {
+                    this.setState({serverError: error.message});
+                }
+                return;
+            }
+
+            this.setState({
+                itemState: MarketplacePluginStatus.INSTALLED,
+                serverError: null,
+            });
+
+            this.props.onInstalled();
+        }
+
+        onInstall = () => {
+            this.setState({installing: true});
+            this.installPlugin(false);
+        }
+
+        render() {
+            const ariaLabel = `${this.props.name}, ${this.props.description}`.toLowerCase();
+            const versionLabel = `(${this.props.version})`;
+
+            let button = (
```

Comment:
would it be worth turning this into its own internal component, that resides in this file and is not exported?

Reply:
Not sure, it handles several props and local state that would need to be passed down and IMO not much gained on punting in it's own internal component.

Reply:
fair, can we at least move it out of the render function? and turn it into it's own?

Reply:
Done!